daemon/container_operations_unix.go

```diff
@@ -610,7 +610,25 @@ func (daemon *Daemon) getNetworkSandbox(container *container.Container) libnetwo
 // ConnectToNetwork connects a container to a network
 func (daemon *Daemon) ConnectToNetwork(container *container.Container, idOrName string) error {
 	if !container.Running {
-		return derr.ErrorCodeNotRunning.WithArgs(container.ID)
+		if container.HostConfig.NetworkMode.IsContainer() {
```

Comment:
Instead of branching with a whole bunch of logic, can we move this to a new function?
I also haven't looked yet, but I'm guessing there is probably some duplicated logic here?